wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java

```diff
@@ -474,6 +552,200 @@ public void test_readerLoadBalancing_autocommitFalse() throws SQLException {
     }
   }
 
+  @TestTemplate
+  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  public void test_readerLoadBalancing_switchAutoCommitInTransaction(final Properties props) throws SQLException {
+    ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.set(props, "true");
+    final String url = getUrl();
+    LOGGER.finest("Connecting to url " + url);
+    try (final Connection conn = DriverManager.getConnection(url, this.defaultProps)) {
+
+      final String writerConnectionId = queryInstanceId(conn);
+      LOGGER.finest("writerConnectionId: " + writerConnectionId);
+
+      conn.setReadOnly(true);
+      String readerId;
+      String nextReaderId;
+
+      // Start transaction while autocommit is on (autocommit is implicitly disabled)
+      // Connection should not be switched while inside a transaction
+      Statement stmt = conn.createStatement();
+      stmt.execute("  StarT   TRanSACtion  REad onLy  ; ");
+      readerId = queryInstanceId(conn);
+      nextReaderId = queryInstanceId(conn);
+      assertEquals(readerId, nextReaderId);
+      conn.setAutoCommit(false); // Switch autocommit value while inside the transaction
+      nextReaderId = queryInstanceId(conn);
+      assertEquals(readerId, nextReaderId);
+      conn.commit();
+
+      assertFalse(conn.getAutoCommit());
+      nextReaderId = queryInstanceId(conn);
+      assertNotEquals(readerId, nextReaderId); // Connection should have switched after committing
+
+      readerId = nextReaderId;
+      nextReaderId = queryInstanceId(conn);
+      // Since autocommit is now off, we should be in a transaction; connection should not be switching
+      assertEquals(readerId, nextReaderId);
+      final ReadWriteSplittingSQLException e =
+          assertThrows(ReadWriteSplittingSQLException.class, () -> conn.setReadOnly(false));
+      assertEquals(SqlState.ACTIVE_SQL_TRANSACTION.getState(), e.getSQLState());
+
+      conn.setAutoCommit(true); // Switch autocommit value while inside the transaction
+      stmt = conn.createStatement();
+      stmt.execute("commit");
+
+      assertTrue(conn.getAutoCommit());
+      readerId = queryInstanceId(conn);
+
+      // Autocommit is now on; connection should switch after each execute
+      for (int i = 0; i < 5; i++) {
+        nextReaderId = queryInstanceId(conn);
+        assertNotEquals(readerId, nextReaderId);
+        readerId = nextReaderId;
+      }
+    }
+  }
+
+  @TestTemplate
+  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  public void test_readerLoadBalancing_remainingStateTransitions(final Properties props) throws SQLException {
```

Comment:
Similar question. Is this scenario still valid?

Reply:
Reader load balancing was removed since pausing this ticket, so these can be deleted. thanks